pymatgen/io/vasp/tests/test_sets.py

```diff
@@ -447,6 +450,46 @@ def test_user_potcar_settings(self):
         potcar = vis.potcar
         self.assertEqual(potcar.symbols, ["Fe", "P", "O"])
 
+    def test_valid_magmom_struct(self):
+        # First test the helper function
+        struct = self.structure.copy()
+        get_valid_magmom_struct(structure=struct, inplace=True, spin_mode="v")
+        props = [isite.properties for isite in struct.sites]
+        self.assertEquals(props, [{"magmom": [1.0, 1.0, 1.0]}] * len(props))
+
+        struct = self.structure.copy()
+        get_valid_magmom_struct(structure=struct, inplace=True, spin_mode="s")
+        props = [isite.properties for isite in struct.sites]
+        self.assertEquals(props, [{"magmom": 1.0}] * len(props))
+        struct.insert(0, "Li", [0, 0, 0])
+        get_valid_magmom_struct(structure=struct, inplace=True, spin_mode="a")
+        props = [isite.properties for isite in struct.sites]
+        self.assertEquals(props, [{"magmom": 1.0}] * len(props))
+
+        struct = self.structure.copy()
+        get_valid_magmom_struct(structure=struct, inplace=True, spin_mode="v")
+        struct.insert(0, "Li", [0, 0, 0], properties={"magmom": 10.0})
+        with self.assertRaises(TypeError) as context:
+            get_valid_magmom_struct(structure=struct, inplace=True, spin_mode="a")
+        self.assertTrue("Magmom type conflict" in str(context.exception))
+
+        # Test the behavior of MPRelaxSet to atomacically fill in the missing magmom
+        struct = self.structure.copy()
+        get_valid_magmom_struct(structure=struct, inplace=True, spin_mode="s")
+        struct.insert(0, "Li", [0, 0, 0])
```

Comment:
This is a common way for the magmom error to occur

Reply:
Do you think there's logic we could add to `struct.insert()` to help there? (e.g. look at what site properties are defined, and warn and/or add a default if there's one missing in the inserted site)

Reply:
You mean instead of doing this? So I think we can leave that up to users, technically there is nothing wrong with some missing properties at a few sites.  Ex: I sometimes get an intercollated structure where all the Li have an 'insertion_energy' property